repositories/soldier.repository.ts

```diff
@@ -1,23 +1,38 @@
-import { Soldier } from "../services/schemas/soldier.zschema";
-import { getSoldiersCollection } from "../db_connection";
-import { Filter } from "mongodb";
+import { Filter, Collection } from 'mongodb';
+
+import { getSoldiersCollection } from '../db_connection';
+import { modifyFilterForArraySupport } from './modifiers';
+import { Soldier } from '../services/schemas/soldier.zschema';
+
+let collection: Readonly<Collection<Soldier>>;
+
+(async () => {
+  collection = await getSoldiersCollection();
```

Comment:
Don't do this, just use` getCollection()`

Reply:
this is what I did initially, you asked me to -
`You can save the collection as a global variable. No need to get it everytime`

Reply:
return to inital

Reply:
resolved